src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,104 @@
 package practice;
 
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
+    public static final String COMA_SEPARATOR = ",";
+    public static final int DIVISION_CONSTANT = 2;
+    public static final int SUBTRACTOR = 1;
+    private final Predicate<Integer> checkNumIsOdd = n -> n % DIVISION_CONSTANT != 0;
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        if (numbers == null || numbers.isEmpty()) {
+            throw new RuntimeException("Can't get min value from list: " + numbers);
+        }
+        return numbers
+                .stream()
+                .map(s -> s.split(COMA_SEPARATOR))
+                .flatMap(Arrays::stream)
+                .mapToInt(Integer::parseInt)
+                .filter(n -> !checkNumIsOdd.test(n))
+                .min()
+                .orElseThrow(()
+                        -> new RuntimeException("Can't get min value from list: " + numbers));
+
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        if (numbers == null || numbers.isEmpty()) {
+            return 0D;
+        }
+        return IntStream.range(0, numbers.size())
+                .map(i -> checkNumIsOdd.test(i) ? numbers.get(i) - SUBTRACTOR : numbers.get(i))
+                 .filter(checkNumIsOdd::test)
+                 .average()
+                 .getAsDouble();
```

Comment:
если заменить 45 строку, то можно обойтись без проверки
```suggestion
                 .orElseThrow(NoSuchElementException::new);
```